src/lib/serviceRoutines/postIndividualContextEntity.cpp

```diff
@@ -58,44 +83,87 @@ std::string postIndividualContextEntity
   ParseData*                 parseDataP
 )
 {
+  AppendContextElementRequest*  reqP                  = &parseDataP->acer.res;
+  AppendContextElementResponse  response;
+  std::string                   entityIdFromPayload   = reqP->entity.id;
+  std::string                   entityIdFromURL       = (compV.size() == 3)? compV[2] : "";
   std::string                   entityId;
+  std::string                   entityTypeFromPayload = reqP->entity.type;
+  std::string                   entityTypeFromURL     = ciP->uriParam[URI_PARAM_ENTITY_TYPE];
   std::string                   entityType;
   std::string                   answer;
-  AppendContextElementRequest*  reqP       = &parseDataP->acer.res;
-  AppendContextElementResponse  response;
 
-  response.entity = reqP->entity;
 
-  if (compV.size() == 3)  // /v1/contextEntities/{entityId}
+  //
+  // 01. Check that total input in consistent and correct
+  //
+
+  // 01.01. entityId::id
+  if ((entityIdFromPayload != "") && (entityIdFromURL != "") && (entityIdFromPayload != entityIdFromURL))
+  {
+    std::string error = "entityId::id differs in URL and payload";
+
+    LM_W(("Bad Input (%s)", error.c_str()));
+    response.errorCode.fill(SccBadRequest, error);
+    return response.render(ciP, IndividualContextEntity, "");
+  }  
+  entityId = (entityIdFromPayload != "")? entityIdFromPayload : entityIdFromURL;
+
+  // 01.02. entityId::type
+  if ((entityTypeFromPayload != "") && (entityTypeFromURL != "") && (entityTypeFromPayload != entityTypeFromURL))
   {
-    entityId   = compV[2];
-    entityType = "";
-
-    if ((reqP->entity.id != "") || (reqP->entity.type != "") || (reqP->entity.isPattern != ""))
-    {
-      LM_W(("Bad Input (unknown field)"));
-      response.errorCode.fill(SccBadRequest, "invalid payload: unknown fields");
-      return response.render(ciP, IndividualContextEntity, "");
-    }
+    std::string error = "entityId::type differs in URL and payload";
+
+    LM_W(("Bad Input (%s)", error.c_str()));
+    response.errorCode.fill(SccBadRequest, error);
+    return response.render(ciP, IndividualContextEntity, "");
   }
-  else if (compV.size() == 2)  // /v1/contextEntities
+  entityType = (entityTypeFromPayload != "")? entityTypeFromPayload :entityTypeFromURL;
+
+
+  // 01.03. entityId::isPattern
+  if (reqP->entity.isPattern == "true")
   {
-    entityId   = reqP->entity.id;
-    entityType = reqP->entity.type;
-
-    if ((entityId == "") && (entityType == ""))
-    {
-      LM_W(("Bad Input (mandatory entityId::id missing)"));
-      response.errorCode.fill(SccBadRequest, "invalid payload: mandatory entityId::id missing");
-      return response.render(ciP, IndividualContextEntity, "");
-    }
+    std::string error = "entityId::isPattern in contextUpdate convenience operation";
```

Comment:
Maybe it better to say:

"entityId::isPattern set to true in contextUpdate convenience operation"